### DISCUSSION TITLE
Allow Abbreviated Target Names

### DISCUSSION THREAD

**Discussion by Soroosh129** [lf-lang/lingua-franca#761] [Ideas]
Some languages such as Python and TypeScript have commonly-used abbreviations such as `Py` and `TS`, which are a lot less verbose than the full names. We should perhaps allow for target declarations to use these abbreviations (e.g. `target Py`).

One approach would be to allow all targets to have abbreviations, or have target-specific abbreviations when appropriate. For the former, we can allow the languages' file extension to be used as the target declaration (e.g., `rs` for Rust), as well as the full name. I personally prefer the latter approach (i.e., only allow `Py` and `TS` for now).

There is also the issue of case sensitivity. Should both `target Py` and `target py` be supported? I think the target names should all be case-sensitive (I went back and forth on this a bit) since a target declaration such as `target c` seems to be incorrect.

For now, I propose the following as the list of recognized target lists:
- C
- Cpp
- Rust
- Python or Py
- TypeScript or TS

**Comment by cmnrd** [comment]
I am a bit afraid about ambiguity here. What if we have to targets with common abbreviations that are the same? Since this affects only a single line per file, I think it is not too bad to force the verbose name.
